### PR TITLE
FAQ Section Updated about Watchdog resets

### PR DESCRIPTION
**Pull request by pasko-zh**
Added a FAQ section about watchdogs: 

- A picture showing how they work
- Explanation how and when to feed the watchdogs
- A simple code showing how to disable the SW watchdog

### CODE REVIEW

**Comment by pasko-zh** [comment]
@krzychb thanks and will document more if needed :smile: 

That's fine with me, I am not aq native english speaker.

**Comment by kgero** [comment]
I do not find this clear. In the example, you stop the software watchdog but continue to feed the hardware watchdog with the function named `system_soft_wdt_feed()`, which according to its name I would assume feeds the software watchdog. In addition, you do not explain why we need to manually feed the hardware watchdog when the software watchdog is stopped. Can you clear this up for me and anyone else in the future using this documentation?

**Comment by pasko-zh** [comment]
@krzychb : Sentence shortened.
@kgero : Is it clearer now how and when these dogs :dog: bite? See [here](https://github.com/pasko-zh/Arduino/blob/0421d3869dee53b81e130be91a82306cdddcee83/doc/faq/readme.md).

**Comment by kgero** [comment]
Yes, thank you!

File: doc/faq/readme.md [hunk 1]
```diff
@@ -43,3 +43,111 @@ You will see this issue only if serial upload was not followed by a physical res
 
 Ref. [#1017](https://github.com/esp8266/Arduino/issues/1017), [#1107](https://github.com/esp8266/Arduino/issues/1107), [#1782](https://github.com/esp8266/Arduino/issues/1782)
 
+
+### I'm getting these Watchdog Resets "wdt reset"�How can I avoid them?
+The esp8266 is equipped with two watchdog timers: A software watchdog and a hardware watchdog. If you don't understand why microcontrollers need a watchdog (timer), you may first read [here](http://www.embedded.com/electronics-blogs/beginner-s-corner/4023849/Introduction-to-Watchdog-Timers).
+
+When your sketch is running�either in the `setup` or during the `loop`�you have to make sure to feed the watchdog periodically. Otherwise "it will bite" and the watchdog reset will stop your sketch. The basic operation is illustrated in the following picture:
```
Comment (krzychb) [inline doc/faq/readme.md]: @pasko-zh, I think my comment was not to the point. Sorry, and let me be more specific. I would rephrase:
> Otherwise "it will bite" and the watchdog reset will stop your sketch. 

to:
Otherwise "it will bite" and the watchdog will reset your sketch. 
or:
Otherwise "it will bite" and the watchdog will restart your module. 

I think that "stop sketch" may confuse people, while "restart" or "reset" better explains what the watchdog does if not fed.

Thank you again for contributing to FAQ :+1:

Krzysztof

Comment (pasko-zh) [inline doc/faq/readme.md]: okey... I will change it.

Comment (pasko-zh) [inline doc/faq/readme.md]: Went for "restart" your sketch, which explains it better than stop your sketch

File: doc/faq/readme.md [hunk 2]
```diff
@@ -43,3 +43,111 @@ You will see this issue only if serial upload was not followed by a physical res
...
+**Can I disable the watchdogs?**
+You can disable the software watchdog but you **cannot** disable the hardware watchdog. Furthermore, you **cannot** change the intervals of the watchdogs timers. 
+
+If you want to disable the software watchdog, you have to call the corresponding SDK functions from the `user_interface.h`. For the documentation on these functions see Espressif's ESP8266 Non-OS SDK API Reference (download on [bbs.espressif](http://bbs.espressif.com/)).
+
+ - `system_soft_wdt_stop()`
+ - `system_soft_wdt_feed()`
+ - `system_soft_wdt_restart()`
+
+The name of the function `system_soft_wdt_feed()` is bit confusing: In general it will feed **both** watchdogs, i.e. the software **and** hardware watchdog. If you disable the software watchdog by calling `system_soft_wdt_stop()` and then call `system_soft_wdt_feed()` afterwards, it will still feed both watchdogs. However, since you have just disabled the software watchdog before, it will have no effect on the software watchdog (timer) and will only affect the hardware watchdog (timer). Thus, from an "End-User's" perspective, this looks like `system_soft_wdt_feed()` feeds the hardware watchdog only.
```
Comment (igrr) [inline doc/faq/readme.md]: Personally I think is in part a bug, in part bad design. Maybe we should rather document how to feed the hardware watchdog via a register?

File: doc/faq/readme.md [hunk 3]
```diff
@@ -43,3 +43,111 @@ You will see this issue only if serial upload was not followed by a physical res
...
+}
+void loop() {
+  Serial.println("Getting busy in 3 seconds...");
+  delay(3000);
+  Serial.println("now!");
+  system_soft_wdt_stop(); // Disable Software watchdog
+  system_soft_wdt_feed(); // Feed Hardware Watchdog
+  for (j = 1; j < 20; j++) {
+    a = micros();
+    busy_1second();
```
Comment (igrr) [inline doc/faq/readme.md]: Can this be replaced with a more intuitive `delayMicroseconds(1000000);`?

File: doc/faq/readme.md [hunk 4]
```diff
@@ -43,3 +43,111 @@ You will see this issue only if serial upload was not followed by a physical res
...
+
+When your sketch is running�either in the `setup` or during the `loop`�you have to make sure to feed the watchdog periodically. Otherwise "it will bite" and the watchdog will restart your sketch. The basic operation is illustrated in the following picture:
+![wdt basic operation](pictures/wdt_basic_operation.png)
+
+
+**How does the esp8266 Arduino Core feed the watchdog(s)?**
+From the perspective of your Arduino sketch, the basic feeding is implicit: Everytime an iteration of your main loop starts again, i.e. it hits the `loop` statement both watchdogs are fed; cf. (1) in the figure above. Esp8266 Arduino libraries, especially core libraries like for instance the `ESP8266WiFi` library take care of feeding the watchdogs, too. Thus, normally you don't have to feed them explicitly.
+
+**When and why should I feed the watchdog(s) explicitly?**
+The software watchdog bites exactly every 3.2 seconds, the hardware watchdog every 7�8 seconds. If your sketch contains code which could exceed running for 3.2 seconds you must feed the watchdogs. I.e. you have to feed them explicitly by calling (cf. (1) in the figure above):
```
Comment (igrr) [inline doc/faq/readme.md]: It is probably worth noting that feeding the watchdog is not the only problem if some code runs for more that a second straight. WiFi and TCP/IP stack will not work as expected after this: station mode wifi will disconnect due to beacon timeout, and TCP stack may get flooded with packets.

**Comment by earlephilhower** [comment]
Thanks for your PR, but the core and libraries have changed enough that this PR now has a merge conflict.

Could you merge it manually with the latest core, so we can consider it for future releases?
